packages/contracts/docs/osx/02-how-to-guides/02-plugin-development/04-upgradeable-plugin/03-setup.md

```diff
@@ -108,355 +116,87 @@ contract SimpleStorageBuild1Setup is PluginSetup {
-  function initializeBuild2(
-    IDAO _dao,
-    uint256 _number,
-    address _account
-  ) external reinitializer(2) {
-    __PluginUUPSUpgradeable_init(_dao);
-    number = _number;
-    account = _account;
-  }
-
-  /// @notice Initializes the plugin when the update from build 1 to build 2 is applied.
-  /// @dev The initialization of `SimpleStorageBuild1` has already happened.
-  function initializeFromBuild1(address _account) external reinitializer(2) {
-    account = _account;
-  }
-
-  function storeNumber(uint256 _number) external auth(STORE_PERMISSION_ID) {
-    number = _number;
-  }
-
-  function storeAccount(address _account) external auth(STORE_PERMISSION_ID) {
-    account = _account;
-  }
-}
+As you can see, we have a constructor storing the implementation contract instantiated via the `new` method in the private immutable variable `implementation` to save gas and an `implementation` function to return it.
+
+:::note
+Specifically important for this type of plugin is the `prepareUpdate()` function. Since we don't know the parameters we will require when updating the plugin to the next version, we can't add the `prepareUpdate()` function just yet. However, keep in mind that we will need to deploy new Plugin Setup contracts in subsequent builds to add in the `prepareUpdate()` function with each build requirements. We see this in depth in the ["How to update an Upgradeable Plugin" section](./05-updating-versions.md).
+:::
+
+### 3. Deployment and publishing
+
+Once you're done with your Plugin Setup contract, we'll need to deploy it so we can publish it into the Aragon OSx protocol. You can deploy your contract with a basic deployment script.
+
+Firstly, we'll make sure our preferred network is well setup within our `hardhat.config.js` file, which should look something like:
+
+```js
+import '@nomicfoundation/hardhat-toolbox';
+
+// To find your Alchemy key, go to https://dashboard.alchemy.com/. Infure or any other provider would work here as well.
+const goerliAlchemyKey = 'add-your-own-alchemy-key';
+// To find a private key, go to your wallet of choice and export a private key. Remember this must be kept secret at all times.
+const privateKeyGoerli = 'add-your-account-private-key';
+
+module.exports = {
+  defaultNetwork: 'hardhat',
+  networks: {
+    hardhat: {},
+    goerli: {
+      url: `https://eth-goerli.g.alchemy.com/v2/${goerliAlchemyKey}`,
+      accounts: [privateKeyGoerli],
+    },
+  },
+  solidity: {
+    version: '0.8.17',
+    settings: {
+      optimizer: {
+        enabled: true,
+        runs: 200,
+      },
+    },
+  },
+  paths: {
+    sources: './contracts',
+    tests: './test',
+    cache: './cache',
+    artifacts: './artifacts',
+  },
+  mocha: {
+    timeout: 40000,
+  },
+};
 ```
 
-</details>
-
-In contrast to build 1, build 2 requires two input arguments: `uint256 _number` and `address _account` that we decode from the bytes-encoded input `_data`.
```

Comment:
Are we completely removing the build 1, 2, 3 plugin examples?

Reply:
No, they appear in the subsequent builds section!